pkg/endpoint/endpoint.go

```diff
@@ -959,7 +959,10 @@ func ParseEndpoint(strEp string) (*Endpoint, error) {
 		return nil, fmt.Errorf("failed to parse base64toendpoint: %s", err)
 	}
 
-	if ep.Status == nil {
+	// Checking the CurrentStatuses is not nil, because some cases status can
+	// be not nil but it doesn't have status in there, cilium will die if no
+	// CurrentStatus is not initialized correctly. Reference issue GH-2477
```

Comment:
Can you change the wording a bit?  Its a bit difficult to follow it.
Something on the lines of
```We need to check for nil for both CurrentStatuses and status, since in some use cases, status will be nil and Cilium will eventually error/panic if CurrentStatus is not initialized correctly.```